subsys/net/lib/nrf_cloud/include/nrf_cloud_codec_internal.h

```diff
@@ -134,22 +136,59 @@ int nrf_cloud_dev_status_json_encode(const struct nrf_cloud_device_status *const
 /** @brief Free memory allocated by @ref nrf_cloud_shadow_dev_status_encode */
 void nrf_cloud_device_status_free(struct nrf_cloud_data *status);
 
-/** @brief Free memory allocated by @ref nrf_cloud_rest_fota_execution_decode */
+/** @brief Free memory allocated by @ref nrf_cloud_rest_fota_execution_decode or
+ * @ref nrf_cloud_fota_job_decode
+ */
 void nrf_cloud_fota_job_free(struct nrf_cloud_fota_job_info *const job);
 
+/** @brief Create an nRF Cloud MQTT FOTA job update payload which is to be sent
```

Comment:
Five attributes to the payload. May be hard to split, but still remarkable achievement ;).